Assets/Scripts/4-generation/Carve.cs

```diff
@@ -27,41 +27,59 @@ void Start()
     // Update is called once per frame
     void Update()
     {
-        if (Input.GetButton("X"))
-        {
-            carveTil();
-        }  
+        carveTil();
     }
 
 
+
     /**
      * method that carve Tile 
-     **/
+     */
     void carveTil()
     {
-        Vector3 targetTile = transform.position;
+        int pressed = notPressed;
+        if (Input.GetButton("X"))
+        {
+            pressed = getIndexByMove();
```

Comment:
Remove pressed var, you don't need it. Just compare:
if(getIndexByMove() != notPressed)
{...}
Besides that look very good ! let's merge it